DirectProgramming/DPC++FPGA/Tutorials/Features/loop_fusion/src/loop_fusion.cpp

```diff
@@ -0,0 +1,278 @@
+  try {
+    queue q(selector, dpc_common::exception_handler,
+            property::queue::enable_profiling{});
+
+    buffer buff_1(m_array_1);
+    buffer buff_2(m_array_2);
+
+    event e = q.submit([&](handler &h) {
+      accessor a_1(buff_1, h, write_only, no_init);
+      accessor a_2(buff_2, h, write_only, no_init);
+
+      h.single_task<DefaultNoFusionKernel>([=
+      ]() [[intel::kernel_args_restrict]] {  // NO-FORMAT: Attribute
+        // Different tripcounts, does not fuse by default
+        for (size_t i = 0; i < kDifferentTripCount + 1; i++) {
+          a_1[i % kArraySize] = i % kArraySize;
+        }
+        for (size_t i = 0; i < kTripCount; i++) {
+          a_2[i % kArraySize] = i % kArraySize;
+        }
+
+      });
+    });
+
+    auto kernel_time = KernelRuntime(e);
+
+    // Kernel consists of two loops different trip counts, each with one array
+    // assignment and two modulos.
+    int num_ops_per_kernel = 3 * kDifferentTripCount + 3 * kTripCount;
+    std::cout << "Throughput for kernel without fusion by default with "
+                 "unequally-sized loops: "
+              << ((double)num_ops_per_kernel / kernel_time) << " Ops/ns\n";
+
+  } catch (sycl::exception const &e) {
+    ErrorReport(e);
+  }
+}
+
+// The compiler is explicitly told to fuse the inner loops using the
+// fpga_loop_fuse<>() function. Were this function not used, the loops would not
+// fuse by default, since the trip counts of the loops are different.
+void FusionFunction(FixedArray &m_array_1, FixedArray &m_array_2) {
+  try {
+    queue q(selector, dpc_common::exception_handler,
+            property::queue::enable_profiling{});
+
+    buffer buff_1(m_array_1);
+    buffer buff_2(m_array_2);
+
+    event e = q.submit([&](handler &h) {
+      accessor a_1(buff_1, h, write_only, no_init);
+      accessor a_2(buff_2, h, write_only, no_init);
+
+      h.single_task<FusionFunctionKernel>([=
+      ]() [[intel::kernel_args_restrict]] {  // NO-FORMAT: Attribute
+        sycl::ext::intel::fpga_loop_fuse([&] {
+          // Different tripcounts, does not fuse by default
+          for (size_t i = 0; i < kDifferentTripCount; i++) {
+            a_1[i % kArraySize] = i % kArraySize;
+          }
+          for (size_t i = 0; i < kTripCount; i++) {
+            a_2[i % kArraySize] = i % kArraySize;
+          }
+        });
+      });
+    });
+
+    auto kernel_time = KernelRuntime(e);
+    // Kernel consists of two loops different trip counts, each with one array
+    // assignment and two modulos.
+    int num_ops_per_kernel = 3 * kDifferentTripCount + 3 * kTripCount;
+    std::cout << "Throughput for kernel with a loop fusion function with "
+                 "unequally-sized loops: "
+              << ((double)num_ops_per_kernel / kernel_time) << " Ops/ns\n";
+
+  } catch (sycl::exception const &e) {
+    ErrorReport(e);
+  }
+}
+
+int main() {
+  // Arrays will be populated in kernel loops
+  FixedArray default_fusion_1;
```

Comment:
You don't need to put 1-per line.   Putting _1,_2 on the same line can be OK, and not adding extra blank lines can shorten the code.    The goal here is to just be aware how work-wrap affects readability and choose the right style to maximize readability.